tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/CodeOwnerEntry.cs

```diff
@@ -12,81 +14,105 @@ internal class CodeOwnerEntry
         public List<string> Owners { get; set; } = new List<string>();
 
         public List<string> Labels { get; set; } = new List<string>();
-
-        public static CodeOwnerEntry Parse(string line)
+        public bool IsValid
         {
-            // The format for the Codeowners File will be:
-            // <path> @<owner> ... @<owner> %<label> ... %<label>
-
-            // get rid of tabs
-            line = line.Replace('\t', ' ');
-            // trim the line to get rid of whitespace before and after.
-            line = line.Trim();
-
-            if (string.IsNullOrEmpty(line) || line.StartsWith('#'))
+            get
             {
-                return null;
+                return !string.IsNullOrEmpty(PathExpression);
             }
+        }
+        private static IEnumerable<string> SplitLine(string line)
+        {
+            // Split the line into segments that are delimited by '@', '%' and the end of the string.
 
-            List<string> entries = SplitLine(line).ToList();
-
-            if (!entries.Any())
+            int previousSplit = -1;
+            for (int i = 0; i < line.Length; i++)
             {
-                return null;
+                if (line[i] == '@' || line[i] == '%')
+                {
+                    if (previousSplit != -1)
+                    {
+                        yield return line.Substring(previousSplit, i - previousSplit).Trim();
+                    }
+                    previousSplit = i;
+                }
             }
 
-            CodeOwnerEntry coe = new CodeOwnerEntry
-            {
-                // the first entry is the path/regex
-                PathExpression = entries[0],
-                ContainsWildcard = entries[0].Contains('*')
-            };
+            // add the last entry
+            yield return line.Substring(previousSplit, line.Length - previousSplit).Trim();
+        }
 
-            // remove the '/' from the path
-            if (coe.PathExpression.StartsWith("/"))
+        public override string ToString()
+        {
+            return $"RegEx:{ContainsWildcard} Expression:{PathExpression} Owners:{string.Join(',', Owners)}  Labels:{string.Join(',', Labels)}";
+        }
+
+        internal static void ParseLabels(CodeOwnerEntry entry, string line)
+        {
+            // Parse a line that looks like # PRLabel: %Label, %Label
```

Comment:
Aren't you assuming the line contains PRLabel?

Reply:
Yes -- that check happens before we call this method -- I guess I could move that check in here.